Added the use of cooldowns

sktan opened the pull request:
Implemented feature to call cooldown then wait 300ms before using long-cooldown abilities.
If cooldown isn't ready to be used, then just use the ability without waiting.

sktan committed:
Added the use of cooldowns

Implemented feature to call cooldown then wait 300ms before using long-cooldown abilities.
If cooldown isn't ready to be used, then just use the ability without waiting.

mouseas commented:
Have you tested it? My experience is that you have to wait until the server responds, about 1 second, before triggering the next ability.

sktan commented:
Tested it and will need to be fixed to wait for the server to respond.
Would changing the timer to 1.1-1.5 seconds be too much you reckon?

mouseas commented:
I would say 5 seconds is too much (since it expires by then), 1.5 seconds sounds like a good idea.

sktan committed:
Changed Cooldown trigger delay

from 300ms to 1500 ms (0.3s to 1.5s)

sktan commented:
Updated the timer to 1.5 seconds
Looks as though it still needs some modification in order for the cooldowns to work.

mouseas commented:
Let me know when you've got it working, tested, and it's ready.

win0na commented:
Tell me when you have fixed the issues with your code, and then I'll fix the merge conflicts that were introduced with the filename change.

sktan committed:
Updated cooldown handler

Calls cooldown, then activates spell on next iteration.
Waits 60 seconds if the cooldown spell is about to come back.

sktan commented:
Not 100% tested, but initial test seems to have done it correctly.

Only calls cooldown if more than 60 seconds on cooldown timer and no cooldown ability currently in lane.

sktan committed:
Fixed bug where skills don't work.

Fixed bug where skills dont work due to cooldown function.

sktan commented:
Further tests seem to work.
